src/Lean/Parser/Tactic.lean

```diff
@@ -66,13 +66,55 @@ doing a pattern match. This is equivalent to `fun` with match arms in term mode.
 @[builtin_tactic_parser] def introMatch := leading_parser
   nonReservedSymbol "intro" >> matchAlts
 
-/-- `decide` will attempt to prove a goal of type `p` by synthesizing an instance
-of `Decidable p` and then evaluating it to `isTrue ..`. Because this uses kernel
-computation to evaluate the term, it may not work in the presence of definitions
-by well founded recursion, since this requires reducing proofs.
-```
+/--
+`decide` attempts to prove a goal of type `p` by synthesizing an instance
+of `Decidable p` and then reducing that instance. If it reduces to `isTrue h`,
+then `h` is used to close the goal.
+
+Limitations:
+- The goal is not allowed to have local variables or metavariables.
+  If there are local variables, you can try to `revert` them first.
+- Because this uses kernel reduction to evaluate the term, it may not work in the
+  presence of definitions by well founded recursion, since this requires reducing proofs.
```

Comment:
```suggestion
- Because this uses kernel reduction to evaluate the term, `Decidable` instances defined
   by well-founded recursion may not work, because evaluating them requires reducing proofs.
```

Reply:
I did some rewording and added an extra comment about `Eq.rec` reduction being an issue. (I don't understand it completely, but I think it's because it takes reducing propositions, which the kernel avoids).